openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMResultTest.java

```diff
@@ -0,0 +1,285 @@
+                8,
+                schema
+        );
+    }
+
+    /**
+     * Gets test result schema.
+     *
+     * @return a {@link DatasetSchema}.
+     */
+    private static DatasetSchema getTestResultSchema() {
+        final List<FieldSchema> schema = new ArrayList<>(10);
+        addFields(schema);
+        schema.add(new FieldSchema("bias", 8, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("predictions", 9, new NumericValueSchema(false)));
+
+        return new DatasetSchema(
+                0, // not used
+                schema
+        );
+    }
+
+    /**
+     * Adds fields for both {@link #getTestSchema()} and {@link #getTestResultSchema()}.
+     *
+     * @param schema The {@link List} to add new fields.
+     */
+    private static void addFields(final List<FieldSchema> schema) {
+        schema.add(new FieldSchema("card", 0, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("amount", 1, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("cat1_generator", 2, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("cat2_generator", 3, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("cat3_generator", 4, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("num1_float", 5, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("num2_float", 6, new NumericValueSchema(false)));
+        schema.add(new FieldSchema("num3_float", 7, new NumericValueSchema(false)));
+    }
+
+    /**
+     * Gets the parameters for LightGBM test.
+     *
+     * @return {@link Map} of key and values.
+     */
+    private static Map<String, String> getTestParams() {
+        final Map<String, String> params = new HashMap<>(12, 1);
+        params.put("max_bin", "512");
+        params.put("learning_rate", "0.05");
+        params.put("boosting_type", "gbdt");
+        params.put("objective", "binary");
+        params.put("num_leaves", "10");
+        params.put("verbose", "-1");
+        params.put("min_data", "100");
+        params.put("boost_from_average", "True");
+        params.put("seed", "42");
+        params.put("num_iterations", "100");
+
+        return params;
+    }
+
+    /**
+     * Load the LightGBM utils or nothing will work.
+     * Also changes parameters.
+     */
+    @BeforeClass
+    public static void setupFixture() {
+        LightGBMUtils.loadLibs();
+    }
+
+    /**
+     * Ensures that every train on LightGBM with the provided train dataset produces
+     * the expected <i>class distribution</i> and <i>feature contributions</i>.
+     *
+     * @throws URISyntaxException    In case of error retrieving the data resource path.
+     * @throws IOException           In case of error reading data.
+     * @throws ModelLoadingException In case of error training the model.
+     */
+    @Test
+    public void trainLightgbm() throws URISyntaxException, IOException, ModelLoadingException {
```

Comment:
I wouldn't call this `trainLightgbm` since it tests more than just the training, it also tests the predictions

Reply:
and the explanations